src/btree.c

```diff
@@ -2908,8 +2914,8 @@ btree_pack_loop(btree_pack_req *req,       // IN/OUT
    log_trace_key(tuple_key, "btree_pack_loop (bottom)");
 
    if (req->hash) {
-      platform_assert(req->num_tuples < req->max_tuples);
-      req->fingerprint_arr[req->num_tuples] =
+      platform_assert(btree_pack_can_fit_tuple(req));
+      fingerprint_start(&req->fingerprint)[req->num_tuples] =
```

Comment:
Here's where you will start to see the use of fingerprint object and its accessor / interfaces.